packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -0,0 +1,354 @@
+import {useVisuallyHidden} from '@react-aria/visually-hidden';
+
+interface ColorAreaAria {
+  /** Props for the color area container element. */
+  colorAreaProps: HTMLAttributes<HTMLElement>,
+  /** Props for the color area gradient foreground element. */
+  gradientProps: HTMLAttributes<HTMLElement>,
+  /** Props for the thumb element. */
+  thumbProps: HTMLAttributes<HTMLElement>,
+  /** Props for the visually hidden horizontal range input element. */
+  xInputProps: InputHTMLAttributes<HTMLInputElement>,
+  /** Props for the visually hidden vertical range input element. */
+  yInputProps: InputHTMLAttributes<HTMLInputElement>
+}
+
+/**
+ * Provides the behavior and accessibility implementation for a color wheel component.
+ * Color wheels allow users to adjust the hue of an HSL or HSB color value on a circular track.
+ */
+export function useColorArea(props: AriaColorAreaProps, state: ColorAreaState, inputXRef: RefObject<HTMLElement>, inputYRef: RefObject<HTMLElement>, containerRef: RefObject<HTMLElement>): ColorAreaAria {
+  let {
+    isDisabled
+  } = props;
+  let formatMessage = useMessageFormatter(intlMessages);
+
+  let {addGlobalListener, removeGlobalListener} = useGlobalListeners();
+
+  let {direction, locale} = useLocale();
+
+  let focusedInputRef = useRef<HTMLElement>(null);
+
+  let focusInput = useCallback((inputRef:RefObject<HTMLElement> = inputXRef) => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputXRef]);
+
+  let stateRef = useRef<ColorAreaState>(null);
+  stateRef.current = state;
+  let {xChannel, yChannel} = stateRef.current.channels;
+  let xChannelStep = stateRef.current.xChannelStep;
+  let yChannelStep = stateRef.current.xChannelStep;
+
+  let currentPosition = useRef<{x: number, y: number}>(null);
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      // these are the cases that useMove doesn't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, don't need to stop propagation, useKeyboard will do that for us
+      e.preventDefault();
+      // remember to set this and unset it so that onChangeEnd is fired
+      stateRef.current.setDragging(true);
+      switch (e.key) {
+        case 'PageUp':
+          stateRef.current.incrementY(stateRef.current.yChannelPageStep);
+          focusedInputRef.current = inputYRef.current;
+          break;
+        case 'PageDown':
+          stateRef.current.decrementY(stateRef.current.yChannelPageStep);
+          focusedInputRef.current = inputYRef.current;
+          break;
+        case 'Home':
+          direction === 'rtl' ? stateRef.current.incrementX(stateRef.current.xChannelPageStep) : stateRef.current.decrementX(stateRef.current.xChannelPageStep);
+          focusedInputRef.current = inputXRef.current;
+          break;
+        case 'End':
```

Comment:
Prob something that I'm doing wrong in VoiceOver maybe but I noticed that doing fn+right arrow (End) moves the VoiceOver cursor off the ColorArea thumb and onto the text representing the color code after shifting the thumb right a step.